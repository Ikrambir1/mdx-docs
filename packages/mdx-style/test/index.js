import React from 'react'
import TestRenderer from 'react-test-renderer'
import MDXStyle, { withComponents } from '../src'

const renderJSON = el => TestRenderer.create(el).toJSON()

describe('MDXStyle', () => {
  test('renders', () => {
    expect(() => {
      renderJSON(
        <MDXStyle />
      )
    }).not.toThrow()
  })

  test('renders with css', () => {
    const json = renderJSON(
      <MDXStyle
        css={{
          root: {
            color: 'tomato'
          },
          h1: {
            fontSize: '32px'
          }
        }}>
        <h1>Hello</h1>
      </MDXStyle>
    )
    expect(json).toMatchSnapshot()
  })

  test('adds styles to custom MDX components', () => {
    const components = {
      h1: props => <h1 {...props} id='test' />
    }
    const Heading = withComponents(({
      components,
      ...props
    }) => {
      return React.createElement(components.h1, props)
    })
    const json = renderJSON(
      <MDXStyle
        components={components}
        css={{
          h1: {
            color: 'tomato'
          }
        }}>
        <Heading>Hello</Heading>
      </MDXStyle>
    )
    const [ style, root ] = json
    const [ h1 ] = root.children
    expect(h1.props.className).toBeTruthy()
    expect(h1.props.id).toBe('test')
    expect(style.props.dangerouslySetInnerHTML.__html).toMatch(/color:tomato/)
  })
})
