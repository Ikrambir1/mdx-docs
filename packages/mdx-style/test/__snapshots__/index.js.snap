// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MDXStyle renders with css 1`] = `
Array [
  <style
    dangerouslySetInnerHTML={
      Object {
        "__html": ".x144fp9m{color:tomato}.x7jn1pg{font-size:32px}",
      }
    }
  />,
  <div
    className="x144fp9m"
  >
    <h1>
      Hello
    </h1>
  </div>,
]
`;
